### ISSUE TITLE
test fails: test_diag (testing.test_kerns.TestCoregion) on Ubuntu with latest tf or with tf0.11

### ISSUE THREAD

**Issue by danstowell** [GPflow/GPflow#269]
Running the tests, I get one single failure:

```
test_diag (testing.test_kerns.TestCoregion) ... I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GT 730M, pci bus id: 0000:02:00.0)
WARNING:tensorflow:From /home/dans/dev/github/GPflow/GPflow/param.py:532 in runnable.: initialize_all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Use `tf.global_variables_initializer` instead.
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GT 730M, pci bus id: 0000:02:00.0)
WARNING:tensorflow:From /home/dans/dev/github/GPflow/GPflow/param.py:532 in runnable.: initialize_all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Use `tf.global_variables_initializer` instead.
FAIL
```

(System info as in #268) Not sure how to diagnose further. Should I try without gpu? (Requires tf rebuild.)

**Comment by jameshensman** [comment]
Do you have a slightly more verbose test output? Most of that text corresponds to a tensorflow warning.

**Comment by danstowell** [comment]
http://pastebin.com/HzdG37vZ

**Comment by icouckuy** [comment]
TF 0.11 has been released so TF master probably does not correspond to 0.11 anymore (only 0.11 is supported).

**Comment by alexggmatthews** [comment]
@danstowell thanks for your interest in GPflow. Because of your bespoke setup it is going to be a bit tricky to replicate the issue. I'm happy to try helping debug it remotely. One possibility is a small floating point difference since we are testing for exact equality in the unit test here. 

What do you get if you print the two things that should be equal in that test. i.e diag(K) and Kdiag in line 83 of test_kerns.py ? Do they look similar but not identical?  (which would indicate the floating point comparison is the issue) Or do they look completely different? (which would suggest a difference coming from your version of TensorFlow).

**Comment by danstowell** [comment]
Thanks for the feedback. I've now installed official TF 0.11 from source. Same issue occurs.

The two matrices that are compared look extremely similar. Inspecting the absolute-differences between the matrices, the max is 8.88178e-16 and the median is 0. I think it's a floating point calculation difference.

So I guess that means I personally can ignore the test failure, though whether you want to loosen the test is not my call. Thanks for your help.

**Comment by fujiisoup** [comment]
`self.assertTrue(np.all(np.diag(K) == Kdiag))` in test_diag
should be 
`self.assertTrue(np.all(np.diag(K), Kdiag))`
?

**Comment by jameshensman** [comment]
Almost, should be 
`self.assertTrue(np.allclose(np.diag(K), Kdiag))`.

**Comment by alexggmatthews** [comment]
Fixed in #270.
